### DISCUSSION TITLE
multiplie files while running from pybricksdev

### DISCUSSION THREAD

**Discussion by krzysztofpniak** [pybricks/support#203] [Q&A]
Is it possible to split project to multiple files and run it in pybricks? I tried relative import but I got an error: "not implemented"

**Comment by dlech** [comment]
For Pybricks on EV3, this is possible. And, yes, relative imports are not implemented in MicroPython, so just use regular imports instead.

Multiple files are not supported on other hubs.

**Reply by krzysztofpniak** [reply]
What is the reason they are not supported on the other hubs? Is it pybricksdev or micropython limitation?

**Reply by laurensvalk** [reply]
The basic hubs (City Hub, Move Hub, Technic Hub) do not have any form of file storage, other than the firmware itself.

The Prime Hub and Inventor Hub have 32MB of storage, so storing files/projects will be possible on those eventually.

We haven't implemented support for this file system yet, as we are prioritizing those features that are applicable to all hubs (MicroPython, motors, sensors, lights, internal sensors, buttons, bluetooth).

**Reply by krzysztofpniak** [reply]
What about some kind of bundling multiple files into one mpy? I'm not sure if python has default tools for this. I can help look for solutions implement it.

**Reply by dlech** [reply]
Please see https://github.com/pybricks/pybricksdev/pull/20.

**Comment by laurensvalk** [comment]
We are working on support for this. For discussion and an experimental implementation, please see https://github.com/pybricks/pybricks-micropython/pull/115. We'd love to get your feedback and ideas for this feature.
